app/src/ingest_bem_pdfs.py

```diff
@@ -76,15 +78,71 @@ def _parse_pdf(file: BinaryIO) -> list[EnrichedText]:
     for text in grouped_texts:
         text.id = str(uuid.uuid1())
     assert len(set(text.id for text in grouped_texts)) == len(grouped_texts)
+
     return grouped_texts
 
 
-def enrich_texts(file: BinaryIO) -> list[EnrichedText]:
-    "Placeholder function. Will be implemented for DST-414, probably in a different file."
+def _enrich_texts(file: BinaryIO) -> list[EnrichedText]:
+    unstuctured_elem_list = partition_pdf(file=file, strategy="fast")
+    enrich_text_list = []
+
     outline: list[Heading] = pdf_utils.extract_outline(file)
-    for _heading in outline:
-        pass
-    return []
+    current_headings: list[Heading] = []
+    for element in unstuctured_elem_list:
+        if element.category == "Footer" or element.category == "Header":
+            continue
+
+        # Unstructured fails to categorize the date strings in the header,
+        # so manually check for that and ignore those too
+        if element.category == "UncategorizedText" and re.match(
+            r"^\d{1,2}-\d{1,2}-\d{4}$", element.text
+        ):
+            continue
+
+        if element.category == "Title":
+            current_headings = _get_current_heading(outline, element, current_headings)
+        try:
+            enriched_text_item = EnrichedText(
+                text=element.text,
+                type=TextType(element.category),
+                page_number=element.metadata.page_number,
+                headings=current_headings,
+                id=element._element_id,
+            )
+            enrich_text_list.append(enriched_text_item)
+        except ValueError:
+            logger.warning(
+                f"{element.category} is not an accepted TextType, {element.text}, {element.metadata.page_number}"
+            )
+    return enrich_text_list
+
+
+def _match_heading(
+    outline: list[Heading], heading_name: str, page_number: int | None
+) -> Heading | None:
+    for heading in outline:
+        if heading.pageno == page_number:
+            # account for spacing differences in unstructured and pdfminer parsing
```

Comment:
Oh! Can you provide an example?

Reply:
Is [this](https://github.com/navapbc/labs-decision-support-tool/pull/68/files#diff-ed2bad192257ddbf1660eab275f187aca890343c45cd7131160e9830d773b502R114) the actual output from unstructured?

Reply:
Interestingly enough the output from unstructured was `TIME AND ATTENDANCE REVIEW PROCESS` (without an extra space) from the pdfminer outline function was `TIME AND ATTENDANCE REVIEW  PROCESS` with an extra space (which is how it looks when reading it)

Reply:
I see https://github.com/navapbc/labs-decision-support-tool/blob/03e45cb09d8f44103e5a6705fc17411641bee9ed/app/tests/src/util/test_pdf_utils.py#L29

<img width="606" alt="image" src="https://github.com/user-attachments/assets/26359ece-8929-4c3d-b26e-8932fbbb580a">